rclpy/rclpy/node.py

```diff
@@ -1622,7 +1622,7 @@ def create_client(
 
         client = Client(
             self.context,
-            client_impl, srv_type, srv_name, qos_profile,
+            client_impl, srv_type, client_impl.service_name, qos_profile,
```

Comment:
this changes the current behavior but i think the intention is to store the actual name with implementation. so i think changing into `client_impl.service_name` is the correct to address the bug.